### PR TITLE
ci: Add paths/paths-ignore to GitHub Actions workflows to reduce unnecessary builds

### PR DESCRIPTION
**Pull request by reedloden**
We're running CIFuzz and CodeQL in many cases where it doesn't make sense. We get charged for GitHub Actions minutes used, so let's make sure to only run them when absolutely necessary.

For CIFuzz, only run if a `*.go` file has been modified.
For CodeQL, only run if a path outside of `docs/*` and `rfd/*` has been modified.

### CODE REVIEW

**Review by russjones — APPROVED** [review]

**Review by wadells — APPROVED** [review]

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -5,10 +5,16 @@ on:
     branches:
       - master
       - branch/*
+    paths-ignore:
+      - 'docs/**'
```
Comment (zmb3) [inline .github/workflows/codeql-analysis.yml:9]: You can probably also ignore

- assets
- build.assets
- docker
- dronegen
- examples
- vagrant

File: .github/workflows/cifuzz.yml [hunk 2]
```diff
@@ -7,10 +7,14 @@ on:
     branches:
       - master
       - branch/*
+    paths:
+      - '**.go'
```
Comment (zmb3) [inline .github/workflows/cifuzz.yml:11]: Maybe also skip any .go files under .github/ as we don't implement any fuzzing tests for the GitHub bot.

**Review by zmb3 — APPROVED** [review]

File: .github/workflows/cifuzz.yml (hunk 2 above)
Comment (reedloden) [reply .github/workflows/cifuzz.yml:11]: You can't use both `paths` and `paths-ignore` in the same workflow, unfortunately.

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (reedloden) [reply .github/workflows/codeql-analysis.yml:9]: I see `*.go` files in some of those. Seems like we'd want to scan them, but I'm guessing the default `make` doesn't build them? Probably could set up separate jobs for them so they get scanned.

**Pull request merged**